### DISCUSSION TITLE
Unable to autoindent for python

### DISCUSSION THREAD

**Discussion by speri203** [LazyVim/LazyVim#297] [General]
Hi, I am unable to get autoindent working in python for LazyVim. For example, when writing a function like "def testing():" after pressing enter I would assume it indents like in VSCode or other nvim distros I have tried. But unfortunately it doesn't do anything. In my options I have the following config added.
local opt = vim.opt

opt.autoindent = true
opt.smartindent = true

and I have the python treesitter plugin installed. Any suggestions?

**Comment by dpetka2001** [comment]
There seems to be a problem with treesitter indentation support for python. There are several issues open in the treesitter github repo. Easiest temporary fix i found was
```lua
return {
  {
    "nvim-treesitter/nvim-treesitter",
    opts = {
      indent = { disable = { "python" } },
    },
  },
}
```
Then indentation works again with vim internal rules.

**Reply by folke** [reply]
Yeah, maybe I should make that the default actually

**Reply by dpetka2001** [reply]
At least until the issue is resolved within treesitter, that would be a good idea I believe.

**Reply by folke** [reply]
done https://github.com/LazyVim/LazyVim/commit/329a2daff493abd2bd5c8fedbf0dfa13039d3931

**Reply by dpetka2001** [reply]
@folke You disabled it globally but you can also disable it just for python. I didn't encounter any other issues with indenting incorrectly with treesitter except for python. Of course it's your decision, but you could just `disable = { "python" }` and leave it enabled for everything else.

**Reply by dpetka2001** [reply]
Something like
```lua
indent = { enable = true, disable = {"python"} }
```
maybe? Up to you.

**Reply by folke** [reply]
Good point! Didn't know that was possible. Changed it

**Reply by speri203** [reply]
> Something like
> 
> ```lua
> indent = { enable = true, disable = {"python"} }
> ```
> 
> maybe? Up to you.

Awesome, this worked! Thank you

**Reply by dpetka2001** [reply]
@speri203 Please be aware that this change is not necessary any more in your config file, since it has been integrated by default in LazyVim and behaves the correct way.

**Reply by speri203** [reply]
> @speri203 Please be aware that this change is not necessary any more in your config file, since it has been integrated by default in LazyVim and behaves the correct way.

Understood, thank you for the update!

**Comment by Flimm** [comment]
This is what I did to disable automatically increasing the indentation after a `def` statement or an `if` statement.

Create the file `~/.config/lazyvim/ftplugin/python.lua` and add these contents:

```
vim.cmd('filetype indent off')
vim.opt_local.cinwords = ""

vim.schedule(function()
  -- disable indentexpr after nvim-treesitter sets it
  vim.bo.indentexpr = ""
end)
```

Then restart Neovim.
